File: RestAPI101_Front/flutter_restapi101/lib/models/todo/todo.dart
import 'package:equatable/equatable.dart';
import 'package:flutter/cupertino.dart';
import 'package:json_annotation/json_annotation.dart';

part 'todo.g.dart';

@immutable
@JsonSerializable()
class Todo extends Equatable {
  final int id;

  @JsonKey(defaultValue: 0)
  final int order;

  final bool done;

  final String title;

  @JsonKey(
    name: 'labels',
    defaultValue: []
  )
  final List<int> labelIds;

  @JsonKey(defaultValue:  '')
  final String description;

  Todo({
    required this.id,
    required this.order,
    required this.done,
    required this.title,
    required this.description,
    required this.labelIds
  });

  factory Todo.fromJson(Map<String, dynamic> json) => _$TodoFromJson(json);
  Map<String, dynamic> toJson() => _$TodoToJson(this);

  @override
  List<Object?> get props => [title, description, done];
}

File: RestAPI101.Backend/RestAPI101.Domain/Models/Todo.cs
using System.Collections.Generic;

namespace RestAPI101.Domain.Models
{
    public class Todo : IEntity
    {
        public int Id { get; set; }

        public int Order { get; set; }

        public bool Done { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public User? User { get; set; }

        public HashSet<Label> Labels { get; }

        public Todo(int id, int order, bool done, string title, string description)
        {
            Id = id;
            Order = order;
            Done = done;
            Title = title;
            Description = description;
            Labels = new HashSet<Label>();
        }

        public Todo(bool done, string title, string description)
        {
            Id = 0;
            Order = 0;
            Done = done;
            Title = title;
            Description = description;
            Labels = new HashSet<Label>();
        }
    }
}
